source/slang/slang-lookup.cpp

```diff
@@ -752,6 +752,31 @@ static void _lookUpMembersInValue(
     return _lookUpMembersInType(astBuilder, name, valueType, request, ioResult, breadcrumbs);
 }
 
+// True if the declaration is of a variety that is overloadable (can have multiple 
+// definitions with the same name)
+// 
+// For example functions are overloadable, but variables are not.
+static bool _isDeclOverloadable(Decl* decl)
+{
+    // TODO(JS): Do we need to special case around ConstructorDecl? or AccessorDecl?
+    // It seems not as they are both function-like and potentially overloadable
+
+    // If it's callable, it's a function and so overloadable 
+    if (auto callableDecl = as<CallableDecl>(decl))
```

Comment:
Would be best to turn this into a switch so the code can be easily extended to handle other cases in the future.